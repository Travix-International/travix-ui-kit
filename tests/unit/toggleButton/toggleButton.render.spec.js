import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import ToggleButton from '../../../components/toggleButton/toggleButton';

describe('ToggleButton', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;

    it('should render correct toggleButton component', () => {
      const mods = ['test-test'];
      const wrapper = shallow(
        <ToggleButton
          handleSelect={() => {}}
          items={['Upper deck', 'Lower deck']}
          mods={mods}
        />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should not render correct toggleButton component', () => {
      const mods = ['test-test'];
      const wrapper = shallow(
        <ToggleButton
          handleSelect={() => {}}
          items={['Upper deck']}
          mods={mods}
        />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
