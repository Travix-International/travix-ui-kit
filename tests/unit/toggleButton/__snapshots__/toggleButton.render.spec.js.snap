// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton #render() should not render correct toggleButton component 1`] = `null`;

exports[`ToggleButton #render() should render correct toggleButton component 1`] = `
<ul
  className="ui-toggle-button ui-toggle-button_test-test"
>
  <li
    className="ui-toggle-button__item ui-toggle-button__item_active"
    onClick={[Function]}
  >
    Upper deck
  </li>
  <li
    className="ui-toggle-button__item"
    onClick={[Function]}
  >
    Lower deck
  </li>
</ul>
`;
